libdrgn/dwarf_info_cache.c

```diff
@@ -676,29 +669,26 @@ drgn_compound_type_from_dwarf(struct drgn_dwarf_info_cache *dicache,
 					"libdw could not parse DIE children");
 		goto err;
 	}
-	if (capacity != num_members) {
-		/* We don't care if this fails. */
-		drgn_type_realloc(&type, num_members,
-				  sizeof(struct drgn_type_member));
-	}
+	type_member_vector_shrink_to_fit(&members);
+
 
```

Comment:
I think the malloc has to stay earlier because of the 'incomplete type' declarations above:

```
	if (declaration) {
		switch (kind) {
		case DRGN_TYPE_STRUCT:
			drgn_struct_type_init_incomplete(type, tag, lang);
```

let me know if I misunderstand though!

Reply:
I missed that, you're right!